cmd/algocfg/datadir.go

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2019 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+package main
+
+import "os"
+
+var dataDirs []string
+
+func resolveDataDir() string {
```

Comment:
these functions exist elsewhere (cmd/goal/... at least) and maybe it's time to extract them? If not libgoal than some other library of algo-command-utils

Reply:
Yes they do.  I moved in the direction of extracting them in this PR - moving them to distinct files.  I didn't want to incur the extra cost right now.